osu.Game.Tests/Visual/Gameplay/TestScenePlayerScoreSubmission.cs

```diff
@@ -181,7 +181,8 @@ public void TestSubmissionOnFail()
             AddUntilStep("wait for fail", () => Player.GameplayState.HasFailed);
             AddStep("exit", () => Player.Exit());
 
-            AddAssert("ensure failing submission", () => Player.SubmittedScore?.ScoreInfo.Passed == false);
+            AddUntilStep("wait for submission", () => Player.SubmittedScore != null);
+            AddAssert("ensure failing submission", () => Player.SubmittedScore.ScoreInfo.Passed == false);
```

Comment:
Could have just swapped `AddAssert()` for `AddUntilStep()` for the same effect, but sure. The null-propagation would have guaranteed that the assert wouldn't pass until `SubmittedScore` was not null.

Reply:
I've intentionally split it apart so test failures like the above become more clear whether they're happening because a score wasn't submitted yet or the actual score had `Passed = true`.